receiver/httpdreceiver/scraper.go

```diff
@@ -0,0 +1,248 @@
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				addToIntMetric(uptime, labels, i, now)
+			}
+		case "ConnsTotal":
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				addToIntMetric(connections, labels, i, now)
+			}
+		case "BusyWorkers":
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				labels.Insert(metadata.L.WorkersState, pdata.NewAttributeValueString("busy"))
+				addToIntMetric(workers, labels, i, now)
+			}
+		case "IdleWorkers":
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				labels.Insert(metadata.L.WorkersState, pdata.NewAttributeValueString("idle"))
+				addToIntMetric(workers, labels, i, now)
+			}
+		case "Total Accesses":
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				addToIntMetric(requests, labels, i, now)
+			}
+		case "Total kBytes":
+			if i, ok := r.parseInt(metricKey, metricValue); ok {
+				bytes := kbytesToBytes(i)
+				addToIntMetric(traffic, labels, bytes, now)
+			}
+		case "Scoreboard":
+			scoreboardMap := parseScoreboard(metricValue)
+			for identifier, score := range scoreboardMap {
+				labels.Upsert(metadata.L.ScoreboardState, pdata.NewAttributeValueString(identifier))
+				addToIntMetric(scoreboard, labels, score, now)
+			}
+		}
+	}
+
+	return rms, nil
+}
+
+// GetStats collects metric stats by making a get request at an endpoint.
+func (r *httpdScraper) GetStats() (string, error) {
+	url := fmt.Sprintf("%s%s", r.cfg.Endpoint, "/server-status?auto")
+	resp, err := r.httpClient.Get(url)
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// parseStats converts a response body key:values into a map.
+func parseStats(resp string) map[string]string {
+	metrics := make(map[string]string)
+
+	fields := strings.Split(resp, "\n")
+	for _, field := range fields {
+		index := strings.Index(field, ": ")
+		if index == -1 {
+			continue
+		}
+		metrics[field[:index]] = field[index+2:]
+	}
+	return metrics
+}
+
+// parseInt converts string to int64.
+func (r *httpdScraper) parseInt(key, value string) (int64, bool) {
+	i, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		r.logInvalid("int", key, value)
+		return 0, false
+	}
+	return i, true
+}
+
+func (r *httpdScraper) logInvalid(expectedType, key, value string) {
+	r.logger.Info(
+		"invalid value",
+		zap.String("expectedType", expectedType),
+		zap.String("key", key),
+		zap.String("value", value),
+	)
+}
+
+// parseScoreboard quantifies the symbolic mapping of the scoreboard.
+func parseScoreboard(values string) map[string]int64 {
+	scoreboard := map[string]int64{
+		"waiting":      0,
+		"starting":     0,
+		"reading":      0,
+		"sending":      0,
+		"keepalive":    0,
+		"dnslookup":    0,
+		"closing":      0,
+		"logging":      0,
+		"finishing":    0,
+		"idle_cleanup": 0,
+		"open":         0,
+	}
+
+	for _, char := range values {
+		switch string(char) {
```

Comment:
Is it worth adding an "unknown" case, to handle a hypothetical future version?

Reply:
Good idea. I added this as the default case, but did not initialize the value in the map. The result is that the 'unknown' count won't show up unless there is actually something unknown. All other values will still default to 0.